stream.go

```diff
@@ -112,6 +112,12 @@ func newClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, meth
 		put func()
 	)
 	c := defaultCallInfo
+	if sc, ok := cc.getMethodConfig(method); ok {
+		c.failFast = !sc.WaitForReady
```

Comment:
I think this code needs to merge the values from the client API and the service config, as per the descriptions of the individual service config fields.  In particular:

- If the client has explicitly specified wait_for_ready via the client API (as opposed to just letting it use the default), that should override the value from the service config.
- For the timeout value, we should use the min of the values specified via the client API and the service config.